src/Init/Data/Nat/Lemmas.lean

```diff
@@ -786,6 +786,16 @@ theorem two_pow_pred_mul_two (h : 0 < w) :
 theorem log2_zero : Nat.log2 0 = 0 := by
   simp [Nat.log2]
 
+@[simp]
+theorem log2_two_pow : (2 ^ n).log2 = n := by
+  match n with
+  | 0 => simp [log2]
+  | n+1 =>
+    unfold log2
+    have if_true : 2 ^ (n + 1) ≥ 2 := Nat.one_lt_two_pow (by simp)
+    simp [if_true]
+    rw [Nat.pow_succ, Nat.mul_div_cancel _ (by simp), log2_two_pow]
```

Comment:
This can be golfed using the results that are already there:
```suggestion
  Nat.eq_of_le_of_lt_succ (by simp [le_log2 (NeZero.ne _)])
    (by simp [log2_lt (NeZero.ne _), Nat.pow_lt_pow_iff_right Nat.one_lt_two]
```

Reply:
Thank you! I addressed it.